apps/06.1Checkers.js

```diff
@@ -8,73 +8,185 @@ var rl = readline.createInterface({
+      // Use that var in .splice() to remove it. But you need to first find the .indexOf it. 
+      // need to use .splice on the grid
+      // the values in .splice() will come from the indexes of position
+      // Position will come from the spot that was jumped
+      // Means that we'll need to check where the piece came from and where in went to
+      // then check the value between those to spots and evalute whether a piece was there or not. 
+      // if true then kill.Checker needs to run. 
+
+} //<--last bracket in Board
 
+function game() {
+  // calls board function to creat a board for .this instance
+  this.board = new Board();
+  // calls two functions to print out the checker and the grid for .this instance
   this.start = function() {
     this.board.createGrid();
-    // Your code here
-  };
-}
+    this.board.createCheckers();
+  //moveChecker recieves arguments from getPrompt(whichPiece, toWhere) --> (start, end)
+  // the arguments are a string but can be read as an array. i.e. [5, 0] and [4, 1]
+  // User input is always a string!!!
+  this.moveChecker = function(start, end){
+    // creating a container the holds the value of the checker the user selects 
+    // takes the index 0 and 1 of start = whichPiece
+    var checker = this.board.selectChecker(start[0], start[1]);
+    // then we need to set that place on the board to null, so we use the same index 0 and 1 of start = null
+    this.board.grid[start[0]][start[1]] = null;
+    // Then we need to move the piece to a new spot. We use the other argument brought in from getPrompt ( , toWhere)
+    // we used the index 0 and 1 of end to find the spot on the grid in .this instance of the board to assign it the value we held
+    // in var checker
+    this.board.grid[end[0]][end[1]] = checker;
+
+    var distance = Math.abs(start[0] - end[0]);
+    // Used the console.logs to figure out what the computer was calculating. Learned I needed to convert the string start and end to Number
+    // console.log("distance is " + distance);
+    // console.log("middle row is " + midRow);
+    // console.log("middle column is " + midColumn);   
+    // console.log(killPosition);   
+    if (distance === 2){
+      // had to create a container for each of these values to store them as numbers to be computed. 
+      var startX = Number(start[0]);
+      var endX = Number(end[0]);
+      var startY = Number(start[1]);
+      var endY = Number(end[1]);
+      // used the midpoint formula to find the average of x and y
+      var midRow = (startX + endX) / 2;
+      var midColumn = (startY + endY) / 2;
+      // created a new container to hold the value of the averages of x and y
+      var killPosition = [midRow, midColumn]; 
+      //  Then we send the value to the killChecker function; again as lon as the distance is 2
+      // Could come back later and make legal moves...
+      this.board.killChecker(killPosition); 
```

Comment:
ALL test passing! Good job leaving lots of comments too.

![wizard](https://i.imgflip.com/1kuiij.jpg)